Content.Server/SimpleStation14/BloodstreamFiller/Systems/BloodstreamFillerSystem.cs

```diff
@@ -0,0 +1,263 @@
+            BreakOnDamage = true,
+            BreakOnTargetMove = true,
+            BreakOnUserMove = true,
+            MovementThreshold = 0.2f,
+            CancelDuplicate = true
+        });
+
+        if (!overfill)
+        {
+            _popup.PopupCursor(Loc.GetString(fillComp.UsePopup, ("target", target)), user);
+            _popup.PopupEntity(Loc.GetString(fillComp.UsedOnPopup), target, target, PopupType.Medium);
+        }
+        else
+        {
+            _popup.PopupCursor(Loc.GetString(fillComp.TargetOverfillPopup, ("target", target)), user, PopupType.MediumCaution);
+            _popup.PopupEntity(Loc.GetString(fillComp.OverfilledPopup), target, target, PopupType.LargeCaution);
+        }
+    }
+
+    private void OnDoAfter(EntityUid uid, BloodstreamComponent component, BloodstreamFillerDoAfterEvent args)
+    {
+        if (args.Cancelled)
+        {
+            return;
+        }
+
+        if (args.Handled || args.Args.Target == null)
+            return;
+
+        if (!TryComp<BloodstreamFillerComponent>(args.Args.Used, out var fillComp))
+        {
+            DebugTools.Assert("Filler component not found");
+            Logger.ErrorS("bloodfiller", $"Filler component not found on entity {ToPrettyString(args.Args.Used.Value)}");
+
+            return;
+        }
+        if (!EntityManager.TryGetComponent<BloodstreamComponent>(args.Args.Target, out var bloodComp))
+        {
+            DebugTools.Assert("Bloodstream component not found");
+            Logger.ErrorS("bloodfiller", $"Bloodstream component not found on entity {ToPrettyString(args.Args.Target.Value)}");
+
+            return;
+        }
+
+        if (!args.Overfill)
+            Fill(args.Args.Target.Value, args.Args.Used!.Value, fillComp, bloodComp);
+        else
+            Overfill(args.Args.User, args.Args.Target.Value, args.Args.Used!.Value, fillComp, bloodComp);
+
+        args.Handled = true;
+    }
+
+    private void Fill(EntityUid target, EntityUid filler, BloodstreamFillerComponent fillComp, BloodstreamComponent bloodComp)
+    {
+        if (!_solution.TryGetSolution(filler, fillComp.Solution!, out var fillerSolution))
+            return;
+
+        var tansfAmount = FixedPoint2.Min(bloodComp.BloodSolution.AvailableVolume, Math.Min((float) fillerSolution.Volume, fillComp.Amount));
+
+        if (tansfAmount > 0)
+        {
+            var drained = _solution.SplitSolution(filler, fillerSolution, tansfAmount);
+
+            _bloodstream.TryModifyBloodLevel(target, drained.Volume, bloodComp);
+
+            _audio.PlayPvs(fillComp.RefillSound, filler);
+        }
+    }
+
+    private void Overfill(EntityUid user, EntityUid target, EntityUid filler, BloodstreamFillerComponent fillComp, BloodstreamComponent bloodComp)
+    {
+        if (!_solution.TryGetSolution(filler, fillComp.Solution!, out var fillerSolution))
+            return;
+
+        if (!TryComp<DamageableComponent>(target, out var damageableComp))
+            return;
```

Comment:
can be combined to one
```suggestion
        if (!_solution.TryGetSolution(filler, fillComp.Solution!, out var fillerSolution) ||
            !TryComp<DamageableComponent>(target, out var damageableComp))
            return;
```